File: SquareRoot/SquareRoot.cpp
// SquareRoot.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <iostream>
int SearchRoot(int num)
{
    int n1 = 0;
    int n = 0;
    for (int i = 0; i <= num; ++i)
    {
        n = n1;
        if (n == num)
            return i;
        if (i * i > num)
            return i - 1;
    }

    return n; // fake one to make it compile, would never be reached
}

int main()
{
    while (true)
    {
        int Num = 0;
        std::cout << "Number search square root for: ";
        std::cin >> Num;
        if (Num < 0)
            break;

        std::cout << "The aquare root of " << Num << " is: " << SearchRoot(Num) << std::endl << std::endl;
    }
}

File: Go/SquareRoot.go
package main

import "fmt"

func SearchRoot(num int) int {
	for i := 0; i <= num; i++ {
		if i*i == num {
			return i
		} else if i*i > num {
			return i - 1
		}
	}

	return -1
}

func main() {
	for {
		fmt.Print("The number to search square root for: ")
		num := 0
		fmt.Scan(&num)
		if num < 0 {
			break
		}

		fmt.Printf("Number %d's square root is %d\n", num, SearchRoot(num))
	}
}
